app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -78,9 +116,37 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                     }
                 }
 
-                rankedUsernames = SortUtils.sortByValue(finalRanking);
-                ArrayAdapter<String> adapter = new RankingAdapter(getActivity(),
-                        rankedUsernames.toArray(new String[rankedUsernames.size()]));
+                Account account = Account.getInstance(getActivity().getApplicationContext());
+                String userNameId = account.getUsername();
+
+                // Sort the rankings
+                Integer[] tmp = new Integer[finalRanking.values().size()];
+                Integer[] rankings = (finalRanking.values().toArray(tmp));
```

Comment:
Why not using int[] ?

Reply:
Because Collection.toArray requires an Integer[]